submissions/xT0PERx/OOP exercise/index.js

```diff
@@ -0,0 +1,88 @@
+class Mammal {
+  constructor(name, gender, saying) {
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+  }
+  toString() {
+    return [this.name, this.gender, this.saying].join(";");
+  }
+}
+
+class Animal extends Mammal {
+  constructor(name, gender, saying, legs = 4) {
+    super(name, gender, saying);
+    this.species = "animal";
+    this.legs = legs;
+  }
+  toString() {
+    return [super.toString(), this.species, this.legs].join(";");
+  }
+}
+
+class Human extends Mammal {
+  constructor(name, gender, saying, friend, legs = 2, hands = 2) {
+    super(name, gender, saying);
+    this.species = "human";
+    this.friend = friend;
+    this.legs = legs;
+    this.hands = hands;
+  }
+  toString() {
+    return [
+      super.toString(),
+      this.species,
+      this.legs,
+      this.hands,
+      this.friend,
+    ].join(";");
+  }
+}
+
+class Cat extends Animal {
+  constructor(name, gender, saying, friend) {
+    super(name, gender, saying);
+    this.species = "cat";
+    this.friend = friend;
```

Comment:
Why `species` and `friend` are handled in a way different from handling `name`, `gender`, and `saying`?
When you answer this question, you will find to optimize your code across entire code base.

Reply:
Probably they are no different :) And I must understand that the superclass should do the things that all classes have, and the classes, respectively, differ from the superclass in that they do something new, unlike it. And since I have in all  classes property :_species_, _friend_, _name_,_gender_,_saying_ that means I need to implement them through the superclass. Am I thinking in the right direction?